data/model/quota.py

```diff
@@ -25,29 +26,140 @@
 get_epoch_timestamp_ms = lambda: int(time.time() * 1000)
 
 
-def add_blob_size(repository_id: int, manifest_id: int, blobs: dict):
-    update_sizes(repository_id, manifest_id, blobs, "add")
-
+def update_quota(repository_id: int, manifest_id: int, blobs: dict, operation: str):
+    # If quota management is disabled, mark the total as stale and return
+    if not features.QUOTA_MANAGEMENT:
+        if config.app_config.get("QUOTA_INVALIDATE_TOTALS", True):
+            reset_backfill(repository_id)
```

Comment:
if I am never enabling quota management, this looks like it will always write to the DB for every push. Ideally if there's no quota management, we should not write to the quota tables

Reply:
This is to handle the scenario where quota was disabled and a push/GC happened to the namespace invalidating the current total. Here we mark it stale to indicate that it has to be re-backfilled. Ideally we wouldn't have to do this but i couldn't find another way. The other option would be to re-backfill the entire table when quota is re-enabled but i saw this as an optimization.

Reply:
instead of an `update` can it be modified to be a check and update. check would ideal `select` against an indexed field. Another option is to delete the row and you can check for non existence on an indexed field

Reply:
Doesn't that still write to the quota tables?

Reply:
it will only write once per repo/namespace. With the current logic, `update` gets called every time even if the data is the same. Updates cause a DB write even if the data is the same

Reply:
Couldn't I add those as conditions to the `where` statement? It shouldn't write if those conditions fail correct?

Reply:
yes that should be good